src/fairseq2/recipes/trainer.py

```diff
@@ -585,6 +585,12 @@ def _maybe_restore_state(self) -> None:
 
         log.info("Training restored. Resuming.")
 
+        try:
+            # Sync vllm model before running online training
+            self._unit.maybe_sync_models(force_sync_vllm=True)
+        except AttributeError:
+            log.info("Vllm model not synced.")
```

Comment:
I'm not sure about doing this here. This also relies on `checkpoint_every_n_steps` being divisible by `sync_vllm_model_every_n_steps` for it to make sense.